staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -37,40 +37,58 @@ import (
 	"k8s.io/klog"
 )
 
-// main demonstrates a leader elected process that will step down if interrupted.
+func buildConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig != "" {
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
 func main() {
 	klog.InitFlags(nil)
+
+	var kubeconfig string
+	var leaseLockName string
+	var leaseLockNamespace string
+	var id string
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	// we can use pod name as holder identity
+	flag.StringVar(&id, "id", os.Getenv("POD_NAME"), "the holder identity name")
+	flag.StringVar(&leaseLockName, "lease-lock-name", "example", "the lease lock resource name")
+	flag.StringVar(&leaseLockNamespace, "lease-lock-namespace", "default", "the lease lock resource namespace")
 	flag.Parse()
-	args := flag.Args()
-	if len(args) != 3 {
-		log.Fatalf("requires three arguments: ID NAMESPACE CONFIG_MAP_NAME (%d)", len(args))
+
+	if id == "" {
+		klog.Fatal("unable to get id (missing POD_NAME environment variable, or id flag).")
 	}
 
-	// leader election uses the Kubernetes API by writing to a ConfigMap or Endpoints
-	// object. Conflicting writes are detected and each client handles those actions
+	// leader election uses the Kubernetes API by writing to a LeaseLock object.
```

Comment:
...by writing to a lock object, which can be a LeaseLock object (preferred), a ConfigMap, or an Endpoints (deprecated) object.